File: javascript/easy/subtract_the_product_and_sum.js
// 1281. Subtract the Product and Sum of Digits of an Integer
// https://leetcode.com/problems/subtract-the-product-and-sum-of-digits-of-an-integer/

var subtractProductAndSum = function(n) {
    var digits = n.toString().split("");
    var product = 1;
    var summed = 0;

    for (var i = 0; i < digits.length; i++) {
        product = product * parseInt(digits[i]);
        summed = summed + parseInt(digits[i]);
    }

    return product - summed;
};

console.log(subtractProductAndSum(234));

File: python/easy/subtract_the_product_and_sum.py
# 1281. Subtract the Product and Sum of Digits of an Integer
# https://leetcode.com/problems/subtract-the-product-and-sum-of-digits-of-an-integer/

def subtractProductAndSum(n):
    # split the integer into digits
    digits = [int(i) for i in list(str(n)) ]

    # find the product/sum of all digits
    product, summed = 1, 0
    for i in digits:
        product = product * i 
        summed = summed + i
    
    return product - summed

print(subtractProductAndSum(234))